### PR TITLE
docker: 19.03.11 -> 19.03.12

### PR DESCRIPTION
**Pull request by Frostman**
###### Motivation for this change

https://github.com/docker/docker-ce/releases/tag/v19.03.12

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/91545)

### CODE REVIEW

File: pkgs/applications/virtualization/docker/default.nix [hunk 1]
```diff
@@ -197,9 +197,9 @@ rec {
   };
 
   docker_19_03 = makeOverridable dockerGen {
-    version = "19.03.11";
-    rev = "42e35e61f352e527082521280d5ea3761f0dee50";
-    sha256 = "0q4l4z2bg269hj1dqaq02f84zgp3p8idw1cxczw383cjl0ra8xgq";
+    version = "19.03.12";
+    rev = "48a66213fe1747e8873f849862ff3fb981899fc6";
```
Comment (jonringer) [inline pkgs/applications/virtualization/docker/default.nix:201]: are we not using releases? https://github.com/docker/docker-ce/releases
```suggestion
    rev = "v${version}";
```
would also have to make it `rec`

Comment (Frostman) [reply pkgs/applications/virtualization/docker/default.nix:201]: That makes sense. I'll try it tonight.

**Comment by Frostman** [comment]
@GrahamcOfBorg test docker

**Comment by Frostman** [comment]
@jonringer I've updated the PR. It works fine and passes tests, but I'm new to Nix, so, not sure if it's the desired way of using `rec`.

`nixosTests.docker` is passing for the PR.

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
Thanks! Ported to stable as 0c486cfadc1b661fdefe5c5c1a7f3915f51d47c0, b3d20c3d5039046ddc29c6b8594fd77b59a931ea

**Pull request merged**